### PR TITLE
scroll after new log line

### PR DESCRIPTION
**Pull request by pajlow**
Scrolls after every new line added to the TextView.

### CODE REVIEW

**Review by Logarythms — APPROVED** [review]

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt [hunk 1]
```diff
@@ -192,6 +192,8 @@ class MainActivity : AppCompatActivity() {
                     }
 
                     _textViewLog?.append(lineToLog + "\n")
```
Comment (christopherstock) [inline app/src/main/java/com/example/pwmobilesdk/MainActivity.kt]: I'd suggest to add one new method `logToConsole` that appends the TextViewLog and invokes the `scrollTextViewLog` at the end of it's method body. This way, `scrollTextViewLog ` would only need to be invoked in one central place. :)

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt [hunk 2]
```diff
@@ -58,6 +57,7 @@ class MainActivity : AppCompatActivity() {
         appBarConfiguration = AppBarConfiguration(navController.graph)
         setupActionBarWithNavController(navController, appBarConfiguration)
 
+        this._scrollViewTextLog      = findViewById(R.id.textview_log_scroll_view)
```
Comment (christopherstock) [inline app/src/main/java/com/example/pwmobilesdk/MainActivity.kt:60]: I've now also checked different solutions and came to the conclusion, that the most simple solution would be to set this general behaviour on refering+initializing the `_scrollViewTextLog` in the following way:
```
Just add to your TextView:
android:gravity="bottom"
android:scrollbars = "vertical"

and set movement method:
myTextView.setMovementMethod(new ScrollingMovementMethod())
```
I've found this solution here:
https://stackoverflow.com/questions/8620153/how-to-get-the-android-textview-to-scroll-down-to-the-end-automatically

I think we can remove the explicit scroll method `scrollTextViewLog` in this way. :)

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt (hunk 1 above)
Comment (christopherstock) [reply app/src/main/java/com/example/pwmobilesdk/MainActivity.kt:194]: Update: I'd recommend to extract one new method `logToConsole` anyway -- even if the `scrollTextViewLog` will be removed as the scroll-movement can be set as a general behaviour of the ScrollView in the `MainActivity.onCreate` method :)

Comment (christopherstock) [reply app/src/main/java/com/example/pwmobilesdk/MainActivity.kt]: After output of `No valid amount provided\n`, the `scrollTextViewLog` needs to be invoked explicitly. :)

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt [hunk 3]
```diff
@@ -262,10 +264,19 @@ class MainActivity : AppCompatActivity() {
...
                     }
+                    scrollTextViewLog()
                 }
             })
     }
 
+    fun scrollTextViewLog() {
+        _scrollViewTextLog?.let {
+            if (it.canScrollVertically(View.FOCUS_DOWN)) {
+                it.fullScroll(View.FOCUS_DOWN)
```
Comment (christopherstock) [inline app/src/main/java/com/example/pwmobilesdk/MainActivity.kt]: Body of scrollTextViewLog can be simplifed:
`_scrollViewTextLog?.fullScroll(View.FOCUS_DOWN)`

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt [hunk 4]
```diff
@@ -262,10 +264,19 @@ class MainActivity : AppCompatActivity() {
                     } else {
                         // Allow your merchant to try another transaction
                     }
+                    scrollTextViewLog()
                 }
             })
     }
 
+    fun scrollTextViewLog() {
+        _scrollViewTextLog?.let {
```
Comment (christopherstock) [inline app/src/main/java/com/example/pwmobilesdk/MainActivity.kt]: I think that the Scroll method invocation for `_scrollViewTextLog` can be simplified. :) 
maybe we can try the shortform too? -- just a one liner inside the method body: :)
```
_scrollViewTextLog?.fullScroll(View.FOCUS_DOWN);
```

File: app/src/main/java/com/example/pwmobilesdk/MainActivity.kt (hunk 3 above)
Comment (christopherstock) [reply app/src/main/java/com/example/pwmobilesdk/MainActivity.kt]: The first item seems to be obsolete as the current solution in the branch is working also with
```
android:fillViewport="true"
```
:)
I'll remove the 1st item on this attribute. :)

**Review by christopherstock — CHANGES_REQUESTED** [review]

**Review by christopherstock — APPROVED** [review]

**Pull request merged**